src/event-stream/event-server.ts

```diff
@@ -678,6 +680,34 @@ export type EventStreamServer = net.Server & {
   closeAsync: () => Promise<void>;
 };
 
+export const bnsImportMiddleware = (db: PgWriteStore) => {
+  let bns_names_onchain_imported: boolean = false;
+  let bns_subdomains_imported: boolean = false;
+  return asyncHandler(async (req, res, next) => {
+    const blockMessage: CoreNodeBlockMessage = req.body;
+    const bnsDir = process.env.BNS_IMPORT_DIR;
+    if (!bns_names_onchain_imported || !bns_subdomains_imported) {
+      const configState = await db.getConfigState();
```

Comment:
It looks like this function can be simplified. On first review, it appeared to be performing a sql query on every new block, and had to re-read the function to see a bit of indirection used for those boolean flags to skip the check. 

Is there a reason why this can't just check `blockMessage.block_height === 1` first (an inexpensive check that will only be hit once), then perform the other checks like this sql query?

Reply:
Yes that should be first

Reply:
Going to skip the closure. First check if height = 1 and we have bnsDir. We should only encounter this once. Then query for import states. If we havent imported one or the other, run import functions.